ext/proj4_c_impl/main.c

```diff
@@ -449,17 +501,106 @@ static VALUE method_crs_to_crs_transform(VALUE self, VALUE x, VALUE y, VALUE z)
   return result;
 }
 
+static VALUE method_crs_to_crs_wkt_str(VALUE self)
+{
+  VALUE result;
+  RGeo_CRSToCRSData *crs_to_crs_data;
+  PJ *crs_to_crs_pj;
+  const char *str;
+
+  result = Qnil;
+  TypedData_Get_Struct(self, RGeo_CRSToCRSData, &rgeo_crs_to_crs_data_type, crs_to_crs_data);
+  crs_to_crs_pj = crs_to_crs_data->crs_to_crs;
+  if (crs_to_crs_pj) {
+    const char *const options[] = {"MULTILINE=NO", NULL};
+    str = proj_as_wkt(PJ_DEFAULT_CTX, crs_to_crs_pj, WKT_TYPE, options);
```

Comment:
I guess the str here is issued from some MALLOC, but it is never freed. I think you should free the `str` once you have the ruby string, right?

After reading [the reference](https://proj.org/development/reference/functions.html) I'm not sure... Did you understand that better than me by any chance ? Otherwise I'll run a test program to be sure nothing may leak

Reply:
We should probably test it, but it seems almost all of their references are `const char *` instead of `char *`, which should mean that they're pointing to some value that is defined at compile time.

The compiler gives me a warning and the program crashes if I try to free the strings and sometimes I get messages about double frees being the issue. My guess is that anything that's a `const char *` should not be freed since it's probably pointing to some shared string that is loaded in from the proj db or hardcoded into the proj library.

Reply:
I dug around the PROJ codebase and it indeed seems like these strings are either hardcoded in somewhere or generated once and stored as a member of the `PJ` instance or a sub-member of some other struct that `PJ` holds.

All of these are cleaned up in `proj_destroy`, so this shouldn't be generating any leaks.

Reply:
Ok thanks for verifying!